pyorbital/tests/test_orbit_elements.py

```diff
@@ -0,0 +1,374 @@
+    )
+    a_0 = a_1 * (1 - delta_1 / 3 - delta_1**2 - (134.0 / 81) * delta_1**3)
+    delta_0 = (
+        (3 / 2.0)
+        * (CK2 / a_0**2)
+        * (
+            (3 * np.cos(np.deg2rad(tle.inclination)) ** 2 - 1)
+            / (1 - tle.excentricity**2) ** (3 / 2)
+        )
+    )
+    expected_sma_ae = a_0 / (1 - delta_0)
+    expected_sma_km = expected_sma_ae * XKMPER
+
+    computed_sma_km = orbit.semi_major_axis * XKMPER / AE
+    assert computed_sma_km == pytest.approx(expected_sma_km, abs=0.1)
+
+
+def test_orbital_period_accuracy():
+    """Test orbital period against corrected mean motion (in seconds)."""
+    tle = MockTLE()
+    orbit = OrbitElements(tle)
+
+    corrected_mean_motion = orbit.original_mean_motion  # radians per minute
+    expected_period_sec = (2 * np.pi / corrected_mean_motion) * 60
+    computed_period_sec = orbit.period * 60
+
+    assert computed_period_sec == pytest.approx(expected_period_sec, abs=1.0)
+
+
+def test_velocity_at_perigee_accuracy():
+    """Test orbital velocity at perigee against analytical reference."""
+    tle = MockTLE()
+    orbit = OrbitElements(tle)
+
+    mu = XKE**2 * AE**3
+    a = orbit.semi_major_axis
+    e = orbit.excentricity
+    r_p = a * (1 - e)
+    v_er_per_min = np.sqrt(mu * (1 + e) / r_p)
+    expected_velocity_kms = v_er_per_min * (XKMPER / 60.0)
+
+    computed_velocity_kms = orbit.velocity_at_perigee()
+    assert computed_velocity_kms == pytest.approx(expected_velocity_kms, abs=0.005)
+
+
+def test_velocity_at_apogee_accuracy():
+    """Test orbital velocity at apogee against analytical reference."""
+    tle = MockTLE()
+    orbit = OrbitElements(tle)
+
+    mu = XKE**2 * AE**3
+    a = orbit.semi_major_axis
+    e = orbit.excentricity
+    r_a = a * (1 + e)
+    v_er_per_min = np.sqrt(mu * (1 - e) / r_a)
+    expected_velocity_kms = v_er_per_min * (XKMPER / 60.0)
+
+    computed_velocity_kms = orbit.velocity_at_apogee()
+    assert computed_velocity_kms == pytest.approx(expected_velocity_kms, abs=0.005)
+
+
+def test_position_vector_perigee_accuracy():
+    """Test position vector at perigee (Mean Anomaly = 0°)."""
+    tle = MockTLE()
+    tle.excentricity = 0.001
+    tle.mean_anomaly = 0.0
+    orbit = OrbitElements(tle)
+    pos = orbit.position_vector()
+    expected_x = orbit.semi_major_axis * (1 - orbit.excentricity)
+    expected_y = 0.0
+    assert pos[0] == pytest.approx(expected_x, rel=1e-6)
+    assert pos[1] == pytest.approx(expected_y, abs=1e-8)
+
+
+@pytest.mark.parametrize("mean_anomaly_deg", [0, 90, 180, 270])
+def test_position_vector_varied(mean_anomaly_deg):
+    """Test position vector output across mean anomalies (basic check)."""
+    tle = MockTLE()
+    tle.mean_anomaly = mean_anomaly_deg
+    orbit = OrbitElements(tle)
+    pos = orbit.position_vector()
+    nu = orbit._get_true_anomaly()
+    r_expected = (
+        orbit.semi_major_axis
+        * (1 - orbit.excentricity**2)
+        / (1 + orbit.excentricity * np.cos(nu))
+    )
```

Comment:
Can these values be hardcoded in the parametrization? As it is now it kind of seems redundant to the internals of the code, right?

```python
@pytest.mark.parametrize(
    ("mean_anomaly_deg", "exp"),
    [(0, 0.001), (90, 0.002), ...],
)
def test_position_vector_varied(mean_anomaly_deg, exp):
```

Reply:
no problem